dagshub/mlflow/patch.py

```diff
@@ -0,0 +1,328 @@
+
+        patch = MlflowMonkeyPatch(funcs_to_patch, classes_to_patch, guaranteed_raises)
+        patch.monkey_patch_module(mlflow)
+        _global_mlflow_patch = patch
+
+
+def _resolve_patches(
+    include: Optional[List[str]],
+    exclude: Optional[List[str]],
+    include_classes: Optional[List[str]],
+    patch_top_level: bool,
+    patch_mlflow_client: bool,
+    raise_on_log_model: bool,
+) -> Tuple[List[str], List[str], List[str]]:
+    """
+    Resolve the arguments for patch_mlflow
+
+    Returns:
+        Tuple of:
+        - List of functions that need to be patched
+        - List of classes that need to be patched
+        - List of functions that should raise exceptions if they are in the stack
+    """
+    funcs_to_patch = _default_patch_functions.copy()
+    classes_to_patch = _default_patch_classes.copy()
+    guaranteed_raises = _default_guaranteed_raises.copy()
+
+    if not patch_top_level:
+        for f in _top_level_functions:
+            if f in funcs_to_patch:
+                funcs_to_patch.remove(f)
+
+    if not patch_mlflow_client:
+        for f in _mlflow_client_functions:
+            if f in funcs_to_patch:
+                funcs_to_patch.remove(f)
+        classes_to_patch.remove("mlflow.tracking.client.MlflowClient")
+
+    if include is not None:
+        funcs_to_patch = funcs_to_patch + include
+
+    if include_classes is not None:
+        classes_to_patch = classes_to_patch + include_classes
+
+    if exclude is not None:
+        for f in exclude:
+            if f in funcs_to_patch:
+                funcs_to_patch.remove(f)
+
+    if not raise_on_log_model:
+        if "log_model" in guaranteed_raises:
+            guaranteed_raises.remove("log_model")
+
+    return funcs_to_patch, classes_to_patch, guaranteed_raises
+
+
+def patch_mlflow(
+    include: Optional[List[str]] = None,
+    exclude: Optional[List[str]] = None,
+    include_classes: Optional[List[str]] = None,
+    raise_on_log_model=True,
+    patch_top_level=True,
+    patch_mlflow_client=False,
+):
+    """
+    Patch MLflow functions, making some of them stop raising exceptions, instead logging them to console.
+    This is useful for long runs that have MLflow failing occasionally,
+    so you don't have to restart the run if logging failed.
+
+    By default, all top-level ``mlflow.log_...`` and ``MlflowClient.log_...`` functions are patched.
+
+    Args:
+        include: List of full names of functions you want to patch additionally to the default ones.
+        exclude: List of functions you DON'T want to patch.\
+            Use this if you need to make sure, for example, all ``log_artifact``
+            functions raise an exception on failure:
+
+            .. code-block:: python
+
+                patch_mlflow(exclude=["mlflow.log_artifact", "mlflow.log_artifacts"])
+
+        include_classes: If ``include`` has classes other than the MlflowClient,
+            you need to list the class here, otherwise it won't be patched
```

Comment:
I wonder if the `classes_to_patch` flag isn't just adding unneeded complexity?
Couldn't you scan all the classes you find and patch any included functions you find?